Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -1058,109 +1058,204 @@ theorem comp_familyOfBFamily {o} (f : ∀ a < o, α) (g : α → β) :
 
 /-! ### Supremum of a family of ordinals -/
 
--- Porting note: Universes should be specified in `sup`s.
-
 /-- The supremum of a family of ordinals -/
+
+@[deprecated iSup (since := "2024-08-27")]
 def sup {ι : Type u} (f : ι → Ordinal.{max u v}) : Ordinal.{max u v} :=
   iSup f
 
-@[simp]
+set_option linter.deprecated false in
+@[deprecated (since := "2024-08-27")]
 theorem sSup_eq_sup {ι : Type u} (f : ι → Ordinal.{max u v}) : sSup (Set.range f) = sup.{_, v} f :=
   rfl
 
 /-- The range of an indexed ordinal function, whose outputs live in a higher universe than the
-    inputs, is always bounded above. See `Ordinal.lsub` for an explicit bound. -/
-theorem bddAbove_range {ι : Type u} (f : ι → Ordinal.{max u v}) : BddAbove (Set.range f) :=
-  ⟨(iSup (succ ∘ card ∘ f)).ord, by
-    rintro a ⟨i, rfl⟩
-    exact le_of_lt (Cardinal.lt_ord.2 ((lt_succ _).trans_le
-      (le_ciSup (Cardinal.bddAbove_range.{_, v} _) _)))⟩
-
+inputs, is always bounded above. See `Ordinal.lsub` for an explicit bound. -/
+theorem bddAbove_range {ι : Type u} (f : ι → Ordinal.{max u v}) : BddAbove (Set.range f) := by
+  use (iSup (succ ∘ card ∘ f)).ord
+  rintro a ⟨i, rfl⟩
+  exact (Cardinal.lt_ord.2 ((lt_succ _).trans_le
+    (le_ciSup (Cardinal.bddAbove_range.{u, v} _) _))).le
+
+/-- `le_iSup` whenever the outputs live in a higher universe than the inputs. -/
+protected theorem le_iSup {ι : Type u} (f : ι → Ordinal.{max u v}) : ∀ i, f i ≤ iSup f :=
+  le_ciSup (bddAbove_range f)
+
+set_option linter.deprecated false in
+@[deprecated Ordinal.le_iSup (since := "2024-08-27")]
 theorem le_sup {ι : Type u} (f : ι → Ordinal.{max u v}) : ∀ i, f i ≤ sup.{_, v} f := fun i =>
-  le_csSup (bddAbove_range.{_, v} f) (mem_range_self i)
+  Ordinal.le_iSup f i
 
+/-- `iSup_le_iff'` whenever the outputs live in a higher universe than the inputs. -/
```

Comment:
I don't undersand this docstring. `iSup_le_iff'` doesn't even exist

Reply:
It's actually `ciSup_le_iff'`, I'll fix all of these docstrings.

Reply:
Do you think those docstrings have much value?

Reply:
Yes! These lemmas are only useful with a particular universe arrangement, unlike the conditionally complete lattice lemmas. That's an important thing to note.